packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -416,13 +416,14 @@ export class AmazonLinuxImage extends GenericSSMParameterImage {
     let virtualization: AmazonLinuxVirt | undefined;
     let storage: AmazonLinuxStorage | undefined;
 
-    if (generation === AmazonLinuxGeneration.AMAZON_LINUX_2022) {
+    if (generation === AmazonLinuxGeneration.AMAZON_LINUX_2022 || generation === AmazonLinuxGeneration.AMAZON_LINUX_2023) {
       kernel = AmazonLinuxKernel.KERNEL5_X;
```

Comment:
You'll find that when this is run and deployed, the SSM parameter won't actually resolve due to Amazon Linux 2023 using Kernel 6.1 rather than Kernel 5.10 like Amazon LInux 2022 is currently coded for.

But also, Amazon Linux 2022 will not deploy via this mechanism, during the preview -> general availability phase, the kernel version was bumped from 5.10 to 5.15 (the SSM param for 5.10 does not exist).

There was an attempt to see if we can re-align it along with introducing 2023, which seems to have opened a bigger can of worms.

https://github.com/aws/aws-cdk/pull/24690

Reply:
Thanks for the feedback.

I guess that this PR should be closed or kept on hold until [this](https://github.com/aws/aws-cdk/issues/21011#issuecomment-1372170948) is implemented.

Unless an alignment of `KERNEL5_X` to `5.15` and the creation of `KERNEL6_X` for `Amazon Linux 2023` can be an acceptable workaround for now.